cob_twist_controller/include/cob_twist_controller/inverse_differential_kinematics_solver.h

```diff
@@ -58,33 +58,27 @@ class InverseDifferentialKinematicsSolver
      * kinematics for
      *
      */
-        InverseDifferentialKinematicsSolver(const KDL::Chain& chain, CallbackDataMediator& data_mediator) :
+    InverseDifferentialKinematicsSolver(const KDL::Chain& chain, CallbackDataMediator& data_mediator) :
         chain_(chain),
         jac_(chain_.getNrOfJoints()),
         jnt2jac_(chain_),
         callback_data_mediator_(data_mediator),
         constraint_solver_factory_(data_mediator, jnt2jac_)
     {
-            last_p_in_vec_ = t_Vector6d::Zero();
+        last_p_in_vec_ = t_Vector6d::Zero();
+        this->kinematic_extension_.reset(KinematicExtensionBuilder::create_extension(this->params_));
```

Comment:
The boost::shared_ptr should be reset() when calling the destructor to free memory of the ExtensionObject.